app/Http/Controllers/ProductsController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Service\ProductService;
+use Carbon\Carbon;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
+
+class ProductsController extends Controller
+{
+    private ProductService $productService;
+
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
```

Comment:
https://php.watch/versions/8.0/constructor-property-promotion
Could be refactored as 

```suggestion
    public function __construct(private ProductService $productService) {}
```